src/cpp/src/block_manager.hpp

```diff
@@ -399,15 +364,13 @@ class BlockManager {
         return block_table[block_idx]->is_free();
     }
 
-    void free_sequence_partially_single_runnning_sequence(size_t seq_id, size_t block_num) {
-        // this method is applicable only for groups with single sequences
+    void free_sequence_partially(size_t seq_id, size_t block_num) {
```

Comment:
here during multiple sequences case, we suppose that we can free the same amount of blocks for each sequence, which can be wrong for beam search case...